### PR TITLE
BaseControl: disabled controls are always omitted

### PR DESCRIPTION
**Pull request by martinknor**

### CODE REVIEW

**Comment by dg** [comment]
Please change commit message to something like 'BaseControl: disabled controls are always omitted'

**Comment by chemix** [comment]
ping @martinknor  can you please update commit message

**Comment by dg** [comment]
Thanks

**Comment by enumag** [comment]
@dg Are you aware that this is BC break? I actually used `->setDisabled(true)->setOmitted(false)` once (don't remember where or why though).

**Comment by dg** [comment]
Yes, it can be a BC break. In what case do you use it?

**Comment by enumag** [comment]
@dg It is an input that is disabled by default but sometimes enabled by javascript.

**Comment by pavelkouril** [comment]
What positive behavior will this change bring (and does it justify the BC break)?

Also, @enumag, how do you get around this BC break to keep the same functionality?

**Comment by matej21** [comment]
You can find discussion here: https://gitter.im/nette/nette/cs?at=55c0a9ac7a6037e67c597569 (czech only)
tl;dr; before this, when you called `->setDisabled(TRUE)->setDisabled(FALSE)` then a control was still omitted

**Comment by matej21** [comment]
fixed in #116

**Comment by enumag** [comment]
@pavelkouril I'd probably just reverse the behavior - the field would be enabled by default and disabled by javascript in some cases. PR #116 should remove the BC break though.
